packages/discord.js/src/structures/UncachedGuildMember.js

```diff
@@ -0,0 +1,328 @@
+     */
+    this.permissions = data.permissions;
+
+    /**
+     * The timestamp this member's timeout will be removed
+     * @type {?number}
+     */
+    this.communicationDisabledUntilTimestamp =
+      data.communication_disabled_until && Date.parse(data.communication_disabled_until);
+
+    /**
+     * The timestamp this member's timeout will be removed, as an ISO8601 timestamp
+     * @type {?string}
+     * @deprecated Use {@link UncachedGuildMember#communicationDisabledUntilTimestamp}
+     * or {@link UncachedGuildMember#communicationDisabledUntil} instead
+     */
+    this.communication_disabled_until = data.communication_disabled_until;
+
+    /**
+     * Whether this UncachedGuildMember is a partial (always true, as it is a partial GuildMember)
+     * @type {boolean}
+     * @readonly
+     */
+    this.partial = true;
+  }
+
+  /**
+   * A link to the member's guild avatar.
+   * @param {ImageURLOptions} [options={}] Options for the image URL
+   * @returns {?string}
+   */
+  avatarURL(options = {}) {
+    return this.avatar && this.client.rest.cdn.guildMemberAvatar(this.guildId, this.id, this.avatar, options);
+  }
+
+  /**
+   * A link to the member's guild avatar if they have one.
+   * Otherwise, a link to their {@link User#displayAvatarURL} will be returned.
+   * @param {ImageURLOptions} [options={}] Options for the Image URL
+   * @returns {string}
+   */
+  displayAvatarURL(options) {
+    return this.avatarURL(options) ?? this.user.displayAvatarURL(options);
+  }
+
+  /**
+   * The time this member joined the guild
+   * @type {?Date}
+   * @readonly
+   */
+  get joinedAt() {
+    return this.joinedTimestamp && new Date(this.joinedTimestamp);
+  }
+
+  /**
+   * The time this member's timeout will be removed
+   * @type {?Date}
+   * @readonly
+   */
+  get communicationDisabledUntil() {
+    return this.communicationDisabledUntilTimestamp && new Date(this.communicationDisabledUntilTimestamp);
+  }
+
+  /**
+   * The last time this member started boosting the guild
+   * @type {?Date}
+   * @readonly
+   */
+  get premiumSince() {
+    return this.premiumSinceTimestamp && new Date(this.premiumSinceTimestamp);
+  }
+
+  /**
+   * The member's id
+   * @type {Snowflake}
+   * @readonly
+   */
+  get id() {
+    return this.user.id;
+  }
+
+  /**
+   * The DM between the client's user and this member
+   * @type {?DMChannel}
+   * @readonly
+   */
+  get dmChannel() {
+    return this.client.users.dmChannel(this.id);
+  }
+
+  /**
+   * The nickname of this member, or their user display name if they don't have one
+   * @type {?string}
+   * @readonly
+   */
+  get displayName() {
+    return this.nickname ?? this.user.displayName;
+  }
+
+  /**
+   * The nickname of the member
+   * @name UncachedGuildMember#nick
+   * @type {?string}
+   * @deprecated Use {@link UncachedGuildMember#nickname} instead
+   */
+  get nick() {
+    return this.nickname;
+  }
```

Comment:
Why is this here?

Reply:
To be non-breaking for https://discord-api-types.dev/api/discord-api-types-v10/interface/APIInteractionGuildMember#nick